cli/command/system/version.go

```diff
@@ -109,7 +125,7 @@ func reformatDate(buildTime string) string {
 	return buildTime
 }
 
-func runVersion(dockerCli command.Cli, opts *versionOptions) error {
+func runVersion(dockerCli command.Cli, cmd *cobra.Command, opts *versionOptions) error {
```

Comment:
I think we should not be passing in `cobra.Command` to the run functions. With the other suggestions (refactoring to remove the need to use `KubeCli` to get version, and getting the kube config from a `path string` arg instead of flags this shouldn't be necessary any more and can be reverted.